cotainr/tests/container/patches.py

```diff
@@ -104,3 +104,22 @@ def patch_disable_add_metadata(monkeypatch):
     monkeypatch.setattr(
         cotainr.container.SingularitySandbox, "add_metadata", lambda _: None
     )
+
+
+@pytest.fixture
+def patch_file_creation_outside_container(monkeypatch):
```

Comment:
This would change the behavior in [test_shell_script_append](https://github.com/DeiC-HPC/cotainr/blob/3e16a4b342fca47af8380621974cb34599451993/cotainr/tests/container/test_singularity_sandbox.py#L150), since it tests `add_to_env()` only after running `Path.write_text`, which takes the responsibility of creating the file. Although I'm not opposed to this change.